### PR TITLE
class1-data

### PR DESCRIPTION
**Pull request by mlukso**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

File: 01week/datatypes.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// 1. Write a JavaScript program to display the current day and time.
+
+// Notes: method "Date()" can pull the current date/time
+
+const currentTime = Date();
```
Comment (reneemeyer) [inline 01week/datatypes.js]: This is a variable, I need a function.

File: 01week/datatypes.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+console.log(currentTime)
+
+
+// 2. Write a JavaScript program to convert a number to a string.
+
+// Notes: method "String()" can convert numbers to a string.
+
+const value2 = 100;
+const numberToString = String(value2)
```
Comment (reneemeyer) [inline 01week/datatypes.js]: This is a variable that stores '100'. I need a function to turn any number into a string and return that string.

File: 01week/datatypes.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+console.log(numberToString)
+
+
+// 3. Write a JavaScript program to convert a string to the number.
+
+// Notes: method "Number()" can convert numbers to a number.
+
+const value1 = "10"
+const stringToNumber =  Number(value1);
```
Comment (reneemeyer) [inline 01week/datatypes.js]: This is a variable that stores 10. I need a function that converts ANY string into a number and converts it.

File: 01week/datatypes.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+// Boolean
+// Null
+// Undefined
+// Number
+// NaN
+// String
+
+// Notes: "typeof" can tell us the data type
+
+const input = "test"
```
Comment (reneemeyer) [inline 01week/datatypes.js]: This is a  variable. I need a function that accepts ANY argument and returns it's data type

**Review by reneemeyer — CHANGES_REQUESTED** [review]
Nice job, please address comments above.

**Comment by mlukso** [comment]
@reneedudley I've revised the code per your suggestions! https://github.com/mlukso/javascript-workbook/pull/1/commits/bdc935f25a7f416bffcea9aeed650ff5866a6910

I'll send you an email as well. Thanks!

File: 01week/datatypes.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+const currentTime=()=>{return Date()}
+
+console.log(currentTime())
+
+
+// 2. Write a JavaScript program to convert a number to a string.
+
+// Notes: method "String()" can convert numbers to a string.
+
+const numberToString=(value1)=>{return String(value1)}
```
Comment (reneemeyer) [inline 01week/datatypes.js:14]: This is a one line function so the curly brackets and return are unnecessary. 
```javascript
const currentTime=()=> Date()
```

File: 01week/datatypes.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+console.log(numberToString(100))
+
+
+// 3. Write a JavaScript program to convert a string to the number.
+
+// Notes: method "Number()" can convert numbers to a number.
+
+const stringToNumber=(value2)=>{return Number(value2)}
```
Comment (reneemeyer) [inline 01week/datatypes.js:23]: Same as the first problem.

File: 01week/datatypes.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+// Boolean
+// Null
+// Undefined
+// Number
+// NaN
+// String
+
+// Notes: "typeof" can tell us the data type
+
+const checkType=(value3)=>{return typeof value3}
```
Comment (reneemeyer) [inline 01week/datatypes.js:37]: same as the first problem

File: 01week/datatypes.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+console.log(checkType('test'))
+
+// 5. Write a JavaScript program that adds 2 numbers together.
+
+// Notes: write simple function to return sum of inputs
+
+const sum=(num1, num2)=>{return Number(num1) + Number(num2)}
```
Comment (reneemeyer) [inline 01week/datatypes.js:45]: same as the first problem

**Review by reneemeyer — COMMENTED** [review]
Nice job, see comments below, address changes then merge

**Pull request merged**